docs/DeveloperGuide.md

```diff
@@ -185,75 +185,104 @@ How the Storage component works:
+the `Order` was addressed to. So for instance, to add an order from a client named `Jamie Tan`, the user would need to ensure
+that a `Person` with `Name` `Jamie Tan` existed in the application first.
+
+Another link we thought would make sense to allow for, was to make it possible to tie a `Order` object to related `Task` objects.
+`Task` objects were meant to help users manage their work, and so we felt there should be a way for a user to relate a `Task`
+to a specific `Order` if they wanted to.
+
+In both of these cases, we did not link the classes directly, hence there is no arrow between the `Order` and `Person` class
+and no arrow between the `Order` and `Task` class in the diagram above. Instead we simply made use of the fact that SalesNote
+maintains both a `UniquePersonList` and a `UniqueOrderList`. To relate a `Order` to a `Person`, it is enough to remember the
+`Name` field (in `UniquePersonList`, two `Person` objects with the same `Name` are considered equal). To relate a `Task` to
+a `Order`, we can make use of the fact that `Order` objects have unique `id` fields.
+
+##### Design choices
+A very reasonable alternative one might consider is linking the classes directly. For instance, allowing a `Person`, to
+have a list of `Order` objects related to the `Person`, and a `Order`, to have a list of `Task` objects related to the `Order`.
+This was an alternative method we considered, that would come with a cost in complexity by relating the `Person`,
+`Task` and `Order` objects. We felt that the method we chose that made use of the `UniqueXList` properties and kept the
+classes more distinct better adhered to the Separation of concerns and Law of Demeter principle.
+
+#### Updating related fields in Person class
+In implementing the fields for the `Task` and `Order` object and considering possible feature flaws, we decided to update
+the fields for the `Person` class as well. The original AB3 treated two people as equal only if their names were spelt exactly
+the same, with this being case-sensitive. We decided that multiple clients having the exact same name was rare
+enough that this notion of equality made sense, however, we felt it should apply regardless of case, i.e. a `Person` with `Name` `john doe`
+should be recognised as the same a `Person` with `Name` `JOHN DOE`.
+
+We updated the equality check to account for this, and also updated the input validation for `Name` to allow at most one
+space between blocks of characters (previously `John   Doe` would be different from `John Doe`. We felt this likely to be
+a mistake and should be avoided).
+
+### Add Order feature
+As mentioned above, to add an `Order`, the `Person` the `Order` is addressed to should already be in `SalesNote`. The
+following is a sequence diagram showing the execution of the command:
+
+![AddOrderSequenceDiagram](images/AddOrderSequenceDiagram.png)
```

Comment:
I just realised if you want to include the `hasorder` and `haspersonwithname` methods in the sequence diagram, you have to wrap the `alt` frames around each of them as well.
<img width="728" alt="Screenshot 2021-11-08 at 11 03 54 AM" src="https://user-images.githubusercontent.com/64767959/140677797-42aff1c2-bc7a-4cd1-8198-b063014646ee.png">

Reply:
'an Order' is more correct I believe, but i'm not sure if alt is the correct type of annotation to add here